README.md

```diff
@@ -1,116 +1,208 @@
+* **Description** Semistructured merge is unable to match **methods** or **constructors** that have been renamed (as their identifiers are now different) or deleted (as their nodes aren't created). This handler handles such cases, attempting to match nodes following other rules.
+* **Behavior** For each method or constructor renamed or deleted, we search left's nodes and right's nodes for a match. For this handler, is enough for one of the following conditions to be true for the match to happen:
+(1) have equal bodies; (2) have similar bodies and equal signatures except for the name; (3) one body is substring of the other. If none of the conditions are satisfied, we treat the node as deleted. Finally, for each triple, we apply a decision based on the type of the renaming or deletion:
+    * If both contributions renamed without body changes, we report a conflict if and only if they renamed to different signatures.
+    * If both contributions deleted or renamed with body changes, we immediately report a conflict if they renamed to different signatures. Otherwise, we merge their contents using textual merge.
+    * If one renamed without body changes and the other deleted renamed with body changes, we report a conflict if they renamed to different signatures or if they renamed to equal signatures but the latter included a reference to the renamed method/constructor.
 
-* Merging 3 directories:
+### [New Element Referencing Edited One Handler](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/src/main/java/br/ufpe/cin/mergers/handlers/NewElementReferencingEditedOneHandler.java)
 
-  `java -jar pathto/jFSTMerge.jar -d "mine" "base" "theirs" -o "output"`
+* **Description** If a developer adds a **class member** that references an edited one inside its body by another developer, semistructured merge can change the semantic of the code since it doesn't care about the elements' bodies. Note that structured merge, on the other hand, deals with this naturally. This handler attempts to fix this issue.
+* **Behavior** After detecting such members, we report a conflict if unstructured merge also reported it.
 
-Where *mine*, *base*, *theirs* and *output* are directory paths.
-The attribute -o is optional, if omitted, *theirs* is used as the output directory.
+### [Type Ambiguity Error Handler](https://github.com/guilhermejccavalcanti/jFSTMerge/blob/master/src/main/java/br/ufpe/cin/mergers/handlers/TypeAmbiguityErrorHandler.java)
 
-<!-- 
-For integration with git type the two commands bellow:
+* **Description** Semistructured merge is unable to output conflicts in **import statements**: even if an import has been edited by two different developers, they're treated as different as their identifiers are different. This can introduce a compilation error (if two imports share a same class name) or a behavioral error (a different import than intended). This handler tries to avoid such errors.
```

Comment:
Tira isto: Semistructured merge is unable to output conflicts in **import statements**: ...

tem que explicar antes o que é o identifier de um import!

de novo, nao diz o problema. diz quando o handler é chamado e o que ele faz. verifica se ha novos imports,...